src/HonzaBotner.Discord.Services/EventHandlers/StaffVerificationEventHandler.cs

```diff
@@ -65,29 +59,12 @@ public async Task<EventHandlerResult> Handle(MessageReactionAddEventArgs eventAr
                 return EventHandlerResult.Stop;
             }
 
+            await _roleManager.RevokeRolesPoolAsync(eventArgs.User.Id, RolesPool.Staff);
+
```

Comment:
So this will always remove the roles and readd them?

Reply:
I also removed the check after removing roles.. I hope it doesn't fail too often. If so, I can reimplement it.

Reply:
I don't think so this should be here. It is the responsibility of the oauth2-callback handler, not this event handler...

Reply:
You should only provide links in buttons similar to how we only sent links in DMs.

Reply:
Addressed, there are now 3 different messages being sent to staff members asking for verification:
Not authenticated -> "You need to authenticate first" + auth button
Authenticated, no staff roles ->"Click to add staff roles" + staff button
Authenticated, with staff roles -> "Click to update/remove staff roles" + staff button AND remove roles button